File: app/src/main/java/com/example/sunkai/heritage/ConnectWebService/HandleUser.java
package com.example.sunkai.heritage.ConnectWebService;

/**
 * Created by sunkai on 2017/3/22.
 * 此类封装了关于登录页有关的服务器请求的方法
 */

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import org.ksoap2.serialization.SoapObject;
import org.ksoap2.serialization.SoapSerializationEnvelope;
import org.ksoap2.transport.HttpTransportSE;
import org.xmlpull.v1.XmlPullParserException;

import java.io.IOException;

public class HandleUser extends BaseSetting {
    @NonNull
    public static Boolean Sign_in(String userName, String PassWord) {
        methodName = "Sign_In";
        soapAction = namespace + "/" + methodName;
        SoapObject soapObject = new SoapObject(namespace, methodName);
        soapObject.addProperty("userName", userName);
        soapObject.addProperty("PassWord", PassWord);
        String result = Get_Post(soapObject);
        return success.equals(result);
    }

    public static int User_Regist(String userName, String passWord, String findPasswordQuestion, String findPassWordAnswer) {
        methodName = "User_Regist";
        soapAction = namespace + "/" + methodName;
        SoapObject soapObject = new SoapObject(namespace, methodName);
        soapObject.addProperty("userName", userName);
        soapObject.addProperty("passWord", passWord);
        soapObject.addProperty("findPasswordQuestion", findPasswordQuestion);
        soapObject.addProperty("findPassWordAnswer", findPassWordAnswer);
        String result = Get_Post(soapObject);
        if (success.equals(result)) {
            return 1;
        } else if ("hadUser".equals(result)) {
            return 0;
        } else {
            return -1;
        }
    }

    public static int Get_User_ID(String userName) {
        methodName = "Get_User_ID";
        soapAction = namespace + "/" + methodName;
        SoapObject soapObject = new SoapObject(namespace, methodName);
        soapObject.addProperty("userName", userName);
        String result = Get_Post(soapObject);
        if (result != null && !error.equals(result) && Integer.parseInt(result) > 0) {
            return Integer.parseInt(result);
        } else {
            return -1;
        }
    }

    @Nullable
    public static String Find_Password_Question(String userName) {
        methodName = "Find_Password_Question";
        soapAction = namespace + "/" + methodName;
        SoapObject soapObject = new SoapObject(namespace, methodName);
        soapObject.addProperty("userName", userName);
        String result = Get_Post(soapObject);
        if (result != null && !error.equals(result))
            return result;
        return null;
    }

    public static boolean Check_Question_Answer(String userName, String questionAnswer) {
        methodName = "Check_Question_Answer";
        soapAction = namespace + "/" + methodName;
        SoapObject soapObject = new SoapObject(namespace, methodName);
        soapObject.addProperty("userName", userName);
        soapObject.addProperty("questionAnswer", questionAnswer);
        String result = Get_Post(soapObject);
        return success.equals(result);
    }

    public static boolean Change_Password(String userName, String Password) {
        methodName = "Change_Password";
        soapAction = namespace + "/" + methodName;
        SoapObject soapObject = new SoapObject(namespace, methodName);
        soapObject.addProperty("userName", userName);
        soapObject.addProperty("Password", Password);
        String result = Get_Post(soapObject);
        return success.equals(result);
    }
}

File: app/src/main/java/com/example/sunkai/heritage/ConnectWebService/HandleUser.kt
package com.example.sunkai.heritage.ConnectWebService


import android.util.Base64
import com.example.sunkai.heritage.Activity.LoginActivity.LoginActivity
import okhttp3.FormBody


/*
 * Created by sunkai on 2018/1/30.
 */

object HandleUser : BaseSetting() {
    fun Sign_In(userName: String, userPassword: String): Boolean {
        val builder = FormBody.Builder()
        builder.add("username", userName)
        builder.add("password", userPassword.replace("\n", ""))
        val result = PutPost("$URL/Sign_In", builder.build())
        return if (result != ERROR && result.toInt() > 0) {
            LoginActivity.userID = result.toInt()
            true
        } else {
            false
        }
    }

    fun User_Regist(userName: String, passWord: String, findPasswordQuestion: String, findPassWordAnswer: String, userImage: ByteArray? = null): Int {
        val userImageString = if (userImage == null) {
            ""
        } else {
            Base64.encodeToString(userImage,Base64.DEFAULT)
        }
        println(userImageString)
        val form = FormBody.Builder()
                .add("username", userName)
                .add("password", passWord)
                .add("findPasswordQuestion", findPasswordQuestion)
                .add("findPasswordAnswer", findPassWordAnswer)
                .add("userImage", userImageString).build()
        val result = PutPost("$URL/UserRegist", form)
        return when (result) {
            "1" -> 1
            "0" -> 0
            else -> -1
        }
    }

    fun Find_Password_Question(userName: String): String? {
        val result = PutGet("$URL/FindPassWordQuestion?username=$userName")
        return if (ERROR != result) result else null
    }

    fun Check_Question_Answer(userName: String, questionAnswer: String): Boolean {
        val form = FormBody.Builder().add("username", userName).add("answer", questionAnswer).build()
        val result = PutPost("$URL/CheckQuestionAnswer", form)
        return SUCCESS == result
    }

    fun Change_Password(userName: String, Password: String): Boolean {
        val form = FormBody.Builder().add("username", userName).add("password", Password).build()
        val result = PutPost("$URL/ChangePassword", form)
        return SUCCESS == result
    }

    fun SendFCMToken(userName: String,token:String):Boolean{
        val form=FormBody.Builder().add("userName",userName).add("token",token).build();
        val result=PutPost("$URL/SendFCMToken",form)
        return SUCCESS==result
    }
}